src/libponyc/reach/reach.c

```diff
@@ -824,9 +824,38 @@ static reach_type_t* add_nominal(reach_t* r, ast_t* type, pass_opt_t* opt)
     default: {}
   }
 
+  bool bare = false;
+
+  if(is_bare(type))
+  {
+    bare = true;
+
+    ast_t* bare_method = NULL;
+    ast_t* member = ast_child(ast_childidx(def, 4));
+
+    while(member != NULL)
+    {
+      if((ast_id(member) == TK_FUN) && (ast_id(ast_child(member)) == TK_AT))
+      {
+        // Only one bare method per bare type.
+        pony_assert(bare_method == NULL);
+        bare_method = member;
+      }
+
+      member = ast_sibling(member);
+    }
```

Comment:
Shouldn't there only be a single member in a bare type - the bare method?

Under what circumstances might a bare type have any fields or non-bare methods?